nav/virtual_layers/src/virtual_layer.cpp

```diff
@@ -0,0 +1,146 @@
+#include <virtual_layers/virtual_layer.h>
+#include <pluginlib/class_list_macros.h>
+
+#include <random>
+
+PLUGINLIB_EXPORT_CLASS(virtual_layers::VirtualLayer, costmap_2d::Layer)
+
+using costmap_2d::LETHAL_OBSTACLE;
+using costmap_2d::NO_INFORMATION;
+
+namespace virtual_layers
+{
+
+VirtualLayer::VirtualLayer() {
+  // ros::Time last_updated_ = ros::Time::now();
+  // ROS_INFO("Start");
+}
+
+void VirtualLayer::onInitialize()
+{
+  ros::NodeHandle nh("~/" + name_);
+  current_ = true;
+  default_value_ = NO_INFORMATION;
+  matchSize();
+
+  dsrv_ = new dynamic_reconfigure::Server<costmap_2d::GenericPluginConfig>(nh);
+  dynamic_reconfigure::Server<costmap_2d::GenericPluginConfig>::CallbackType cb = boost::bind(
+      &VirtualLayer::reconfigureCB, this, _1, _2);
+  dsrv_->setCallback(cb);
+}
+
+void VirtualLayer::matchSize()
+{
+  Costmap2D* master = layered_costmap_->getCostmap();
+  resizeMap(master->getSizeInCellsX(), master->getSizeInCellsY(), master->getResolution(),
+            master->getOriginX(), master->getOriginY());
+}
+
+
+void VirtualLayer::reconfigureCB(costmap_2d::GenericPluginConfig &config, uint32_t level)
+{
+  enabled_ = config.enabled;
+}
+
+std::vector<std::vector<double>> getRandomPoints(double t) {
```

Comment:
Rename function, I think technically it's not generating random points anymore

Also, it's not a class member (which is fine), but you may want to make it a static non-member function or put it under an anonymous namespace